pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -47,38 +63,49 @@ let
         ${replaced} = v;
         "tree-sitter-${replaced}" = v;
       }
-    ) generatedDerivations;
+    ) parsersWithMeta;
 
-  allGrammars = lib.attrValues generatedDerivations;
+  allGrammars = builtins.attrValues parsersWithMeta;
 
   # Usage:
   # pkgs.vimPlugins.nvim-treesitter.withPlugins (p: [ p.c p.java ... ])
   # or for all grammars:
   # pkgs.vimPlugins.nvim-treesitter.withAllGrammars
   withPlugins =
     f:
+    let
+      selectedGrammars = f (tree-sitter.builtGrammars // builtGrammars);
+
+      grammarPlugins = map grammarToPlugin selectedGrammars;
+
+      queryPlugins = lib.pipe selectedGrammars [
+        (map (g: g.passthru.associatedQuery or null))
+        (builtins.filter (q: q != null))
+      ];
```

Comment:
> `withAllGrammars` includes every query and every parser currently, to be backwards-compatible

How do you install all grammars without queries then?

Reply:
> > `withAllGrammars` includes every query and every parser currently, to be backwards-compatible
> 
> How do you install all grammars without queries then?

Does nvim-treesitter install grammars without queries?

Having queries installed isn't necessarily "disruptive"; they don't get loaded and impact startup time in a meaningful way. They only get loaded when needed, e.g. if the user enables tree-sitter based syntax highlighting for a buffer.

Reply:
Oh we install queries from the nvim-treesitter repo, nvm then

Reply:
> How do you install all grammars without queries then?

To install just the parsers, is just add `vimPlugins.nvim-treesitter.parsers.parser` to the plugins. Also, I am pretty sure a parser without queries would be useless